packages/ra-supabase-ui-materialui/src/guessers/inferElementFromType.ts

```diff
@@ -29,22 +33,82 @@ export const inferElementFromType = ({
         hasType('reference', types)
     ) {
         const reference = description.split('`')[1].split('.')[0];
-        return new InferredElement(types.reference, {
-            source: name,
-            reference,
-            ...props,
-        });
+        const referenceResourceDefinition = schema.definitions?.[reference];
+        if (!referenceResourceDefinition) {
+            throw new Error(
+                `The referenced resource ${reference} is not defined in the API schema`
+            );
+        }
+        const referenceRecordRepresentationField =
+            inferRecordRepresentationField(referenceResourceDefinition);
+
+        return new InferredElement(
+            types.reference,
+            {
+                source: name,
+                reference,
+                ...props,
+            },
+            hasType('autocompleteInput', types) &&
+            referenceRecordRepresentationField
+                ? [
+                      new InferredElement(
+                          types.autocompleteInput,
+                          referenceRecordRepresentationField
+                              ? {
+                                    optionText:
+                                        referenceRecordRepresentationField,
+                                }
+                              : {}
+                      ),
+                  ]
+                : undefined,
+            hasType('autocompleteInput', types) &&
+            !referenceRecordRepresentationField
+                ? `Could not infer the field to use to filter referenced ${reference} records. Please provide the \`filterToQuery\` prop to the <AutocompleteInput> component. See https://github.com/marmelab/ra-supabase/blob/main/packages/ra-supabase/README.md#autocompleteinput-with-references`
+                : undefined
+        );
     }
     if (
         name.substring(name.length - 4) === '_ids' &&
         hasType('reference', types)
```

Comment:
not 100% sure but shouldn't you check for `hasType('referenceArray', types)` here instead? :thinking: